parachain/pallets/erc20-app/src/lib.rs

```diff
@@ -131,7 +132,16 @@ impl<T: Config> Application for Module<T> {
 		Self::handle_event(payload_decoded)
 	}
 
-	fn address() -> H160 {
-		Address::get()
+}
+#[derive(Copy, Clone)]
+struct Proxy<T: Config>(PhantomData<T>);
+
+impl<T: Config> Application for Proxy<T> {
+	fn handle(&self, payload: &[u8]) -> DispatchResult {
+		<Module<T>>::handle(payload)
 	}
 }
+
+pub fn make_proxy<T: Config>() -> Box<dyn Application> {
```

Comment:
We need to create a proxy object as pallets can't be instantiated and passed around like normal objects.